crates/hir-def/src/body/lower.rs

```diff
@@ -60,6 +62,13 @@ pub(super) fn lower(
     krate: CrateId,
     is_async_fn: bool,
 ) -> (Body, BodySourceMap) {
+    // We cannot leave the root span map empty and let any identifier from it be treated as root,
+    // because when inside nested macros `SyntaxContextId`s from the outer macro will be interleaved
+    // with the inner macro, and that will cause confusion because they won't be the same as `ROOT`
+    // even though they should be the same. Also, when the body comes from multiple expansions, their
+    // hygiene is different.
+    let span_map =
+        expander.current_file_id().macro_file().map(|file_id| db.expansion_span_map(file_id));
```

Comment:
The expander contains a `span_map: OnceCell<SpanMap>`, we can populate that in this case